src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala

```diff
@@ -750,53 +716,109 @@ object HealthCheckConsensus {
     implicit val checkedPeerCodec: Codec[CheckedPeer] = deriveCodec
   }
 
-  case class ConsensusHealthStatus(
+  sealed trait ConsensusHealthStatus[+K <: HealthCheckKey, +A <: HealthCheckStatus] {
+    val key: K
+    val checkedPeer: CheckedPeer
+    val checkingPeerId: Id
+    val roundId: HealthcheckRoundId
+    val status: A
+    val clusterState: Map[Id, NodeState]
+  }
+
+  object ConsensusHealthStatus {
+//    implicit val encodeConsensusHealthStatus: Encoder[ConsensusHealthStatus[HealthCheckStatus]] = deriveEncoder[ConsensusHealthStatus[HealthCheckStatus]]
+//    implicit val decodeConsensusHealthStatus: Decoder[ConsensusHealthStatus[HealthCheckStatus]] = deriveDecoder[ConsensusHealthStatus[HealthCheckStatus]]
+    implicit val consensusHealthStatusEncoder: Encoder[ConsensusHealthStatus[HealthCheckKey, HealthCheckStatus]] =
+      Encoder.instance {
+        case a @ PingHealthStatus(_, _, _, _, _, _)            => a.asJson
```

Comment:
I think we don't need this pattern match if we're doing `_.asJson` in either case.

Reply:
I will check that!